basic-examples/expressions/subworkflow.yaml

```diff
@@ -11,7 +11,7 @@ input:
 steps:
   example:
     plugin:
-      src: quay.io/arcalot/arcaflow-plugin-template-python
+      src: quay.io/arcalot/arcaflow-plugin-template-python:all_d5dd9d6

```

Comment:
Are we _really_ sure that we want to put a tag (other than the implicit `latest`) on this container image?

I think this creates a burden to keep this file up to date which is greater than that imposed by letting the reference "float".

Reply:
We've previously discussed Arcalot's general image version pinning policy. We want to pin our versions to minimize unexpected changes. In this repo, it is especially important that we can expect reproducible behavior. Your suggestion makes the behavior less stable. 

At the very least, there's a trade-off here, that has already been acknowledged and accepted for this repo. Every other workflow here uses versioned images. This workflow is currently the only exception. This isn't the issue to start changing this repo's image version pinning policy, and it feels like an unproductive conversation.

Reply:
First, I concur that this should use a versioned tag and not `:latest`. With our plugins still in v0 releases, we aren't doing work to guarantee compatibility between releases, and using `:latest` will almost certainly lead to drift that will break an example workflow. That said, this should be pointing to a released version of the plugin, not a development tag, since by convention development tags should expire in 3 months.

Second, I question using the template plugin for an example workflow. This would make more sense to me using the example plugin instead. I know we have kind of debated in the past whether we need both a template and an example plugin, but the current state is that we maintain those independently, and the real purpose of the template plugin is to be exactly that, a template for creating a new plugin repo.

Reply:
I used the template plugin because it requires a smaller input footprint, and it doesn't have a regular expression expectation, while the example plugin does.

Reply:
We should probably add a new step in the example plugin that does not use a OneOf nor a regular expression.

Reply:
We should get this off of a version that could expire under our policy.